src/charm.py

```diff
@@ -442,15 +449,16 @@ def _on_relation_changed(self, _: ops.HookEvent, charm_state: CharmState) -> Non
         """
         # the main unit has changed so workers must be restarted
         if self.get_main_unit() != self.unit.name:
-            self.change_config(charm_state)
+            logger.debug("_on_relation_changed emitting reconcile")
+            self.reconcile(charm_state)
         # Reload NGINX configuration with new main address
         nginx_container = self.unit.get_container(synapse.SYNAPSE_NGINX_CONTAINER_NAME)
         if not nginx_container.can_connect():
             logger.warning(
                 "Relation changed received but NGINX container is not available for reloading."
             )
             return
-        pebble.replan_nginx(nginx_container, self.get_main_unit_address())
+        pebble.restart_nginx(nginx_container, self.get_main_unit_address())
```

Comment:
We do a restart now instead of a replan, can you say why?

Reply:
We can discuss in the same thread here: https://github.com/canonical/synapse-operator/pull/378#discussion_r1629410433 :)

Reply:
I'll close this one considering the approvals (and my anxiety to test it). Please, feel free to reach out for further discussion regarding that.